microsoft/testsuites/core/infiniand.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Ubuntu
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    Tests the functionality of infiniband.
+    """,
+)
+class Infiniband(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Determine whether the VM is IB over ND, IB over SR-IOV, or non-HPC
+        2. Ensure waagent is configures with OS.EnableRDMA=y
+        3. Check that appropriate drivers are present
+        """,
+        priority=2,
+    )
+    def verify_hpc_vm(self, log: Logger, node: Node) -> None:
+
+        # Ubuntu is used for HPC SKU but manual configuration required.
+        # It's exceptional.
+        if isinstance(node.os, Ubuntu):
+            raise SkippedException(
+                f"{node.os.name} not supported. "
```

Comment:
Make the exception message is accurate like the comments above. And in this case, if the exception message is clear enough, it doesn't need repeat it in comments.

Reply:
That comment is from v2. I'd like to know whether skipping Ubuntu is still a good choice for v3.  While Ubuntu is not the recommended distro for HCP machines, it is supported and there are HCP Ubuntu images on the Azure market place. This logic may be outdated.

Reply:
Please double check it. If Ubuntu supports HPC, please remove this logic.

Reply:
refer https://techcommunity.microsoft.com/t5/azure-compute/azure-hpc-vm-images/ba-p/977094